Compiler error

It seems that recently GCC has added a new warning, or started catching warnings, etc. The clever hack from #1906 now throws warnings. Tested with GCC 9.3 on Debian Unstable.

```
    gcc -pthread -B /home/kaleb/miniconda3/compiler_compat -Wl,--sysroot=/ -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -Werror -fPIC -DVHPI_CHECKING=1 -DIUS= -Icocotb/share/include -I/home/kaleb/dev/cocotb/.tox/py3/include -I/home/kaleb/miniconda3/include/python3.7m -c cocotb/share/lib/vhpi/VhpiCbHdl.cpp -o build/temp.linux-x86_64-3.7/cocotb/share/lib/vhpi/VhpiCbHdl.o -std=c++11 -Wall -Wextra -Wcast-qual -Wwrite-strings -Wconversion -Wnon-virtual-dtor -Woverloaded-virtual -D__STDC_FORMAT_MACROS
    cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
    cocotb/share/lib/vhpi/VhpiCbHdl.cpp: In lambda function:
    cocotb/share/lib/vhpi/VhpiCbHdl.cpp:940:9: error: assignment from temporary initializer_list does not extend the lifetime of the underlying array [-Werror=init-list-lifetime]
      940 |         }},
          |         ^
    cocotb/share/lib/vhpi/VhpiCbHdl.cpp:946:9: error: assignment from temporary initializer_list does not extend the lifetime of the underlying array [-Werror=init-list-lifetime]
      946 |         }},
          |         ^
    cocotb/share/lib/vhpi/VhpiCbHdl.cpp:957:9: error: assignment from temporary initializer_list does not extend the lifetime of the underlying array [-Werror=init-list-lifetime]
      957 |         }},
          |         ^
    cocotb/share/lib/vhpi/VhpiCbHdl.cpp:969:9: error: assignment from temporary initializer_list does not extend the lifetime of the underlying array [-Werror=init-list-lifetime]
      969 |         }},
          |         ^
    cc1plus: all warnings being treated as errors
    error: command 'gcc' failed with exit status 1
```

--- a/cocotb/share/lib/fli/FliImpl.cpp
+++ b/cocotb/share/lib/fli/FliImpl.cpp
@@ -577,16 +577,20 @@ GpiIterator *FliImpl::iterate_handle(GpiObjHdl *obj_hdl, gpi_iterator_sel_t type
 }
 
 decltype(FliIterator::iterate_over) FliIterator::iterate_over = []{
-    std::initializer_list<FliIterator::OneToMany> region_options;
-    std::initializer_list<FliIterator::OneToMany> signal_options;
-    std::initializer_list<FliIterator::OneToMany> variable_options;
+    std::initializer_list<FliIterator::OneToMany> region_options = {
+        FliIterator::OTM_CONSTANTS,
+        FliIterator::OTM_SIGNALS,
+        FliIterator::OTM_REGIONS,
+    };
+    std::initializer_list<FliIterator::OneToMany> signal_options = {
+        FliIterator::OTM_SIGNAL_SUB_ELEMENTS,
+    };
+    std::initializer_list<FliIterator::OneToMany> variable_options = {
+        FliIterator::OTM_VARIABLE_SUB_ELEMENTS,
+    };
 
     return decltype(FliIterator::iterate_over) {
-        {accArchitecture, region_options = {
-            FliIterator::OTM_CONSTANTS,
-            FliIterator::OTM_SIGNALS,
-            FliIterator::OTM_REGIONS,
-        }},
+        {accArchitecture, region_options},
         {accEntityVitalLevel0, region_options},
         {accArchVitalLevel0, region_options},
         {accArchVitalLevel1, region_options},
@@ -612,16 +616,12 @@ decltype(FliIterator::iterate_over) FliIterator::iterate_over = []{
         {accForGenerate, region_options},
         {accConfiguration, region_options},
 
-        {accSignal, signal_options = {
-            FliIterator::OTM_SIGNAL_SUB_ELEMENTS,
-        }},
+        {accSignal, signal_options},
         {accSignalBit, signal_options},
         {accSignalSubComposite, signal_options},
         {accAliasSignal, signal_options},
 
-        {accVariable, variable_options = {
-            FliIterator::OTM_VARIABLE_SUB_ELEMENTS,
-        }},
+        {accVariable, variable_options},
         {accGeneric, variable_options},
         {accGenericConstant, variable_options},
         {accAliasConstant, variable_options},

--- a/cocotb/share/lib/vhpi/VhpiCbHdl.cpp
+++ b/cocotb/share/lib/vhpi/VhpiCbHdl.cpp
@@ -921,52 +921,52 @@ VhpiNextPhaseCbHdl::VhpiNextPhaseCbHdl(GpiImplInterface *impl) : GpiCbHdl(impl),
 
 decltype(VhpiIterator::iterate_over) VhpiIterator::iterate_over = []{
     /* for reused lists */
-    std::initializer_list<vhpiOneToManyT> root_options;
-    std::initializer_list<vhpiOneToManyT> sig_options;
-    std::initializer_list<vhpiOneToManyT> simplesig_options;
-    std::initializer_list<vhpiOneToManyT> gen_options;
+    std::initializer_list<vhpiOneToManyT> root_options = {
+        vhpiInternalRegions,
+        vhpiSigDecls,
+        vhpiVarDecls,
+        vhpiPortDecls,
+        vhpiGenericDecls,
+        vhpiConstDecls,
+        //    vhpiIndexedNames,
+        vhpiCompInstStmts,
+        vhpiBlockStmts,
+    };
+    std::initializer_list<vhpiOneToManyT> sig_options = {
+        vhpiIndexedNames,
+        vhpiSelectedNames,
+    };
+    std::initializer_list<vhpiOneToManyT> simplesig_options = {
+        vhpiDecls,
+        vhpiInternalRegions,
+        vhpiSensitivitys,
+        vhpiStmts,
+    };
+    std::initializer_list<vhpiOneToManyT> gen_options = {
+        vhpiDecls,
+        vhpiInternalRegions,
+        vhpiSigDecls,
+        vhpiVarDecls,
+        vhpiConstDecls,
+        vhpiCompInstStmts,
+        vhpiBlockStmts,
+    };
 
     return decltype(VhpiIterator::iterate_over) {
-        {vhpiRootInstK, root_options = {
-            vhpiInternalRegions,
-            vhpiSigDecls,
-            vhpiVarDecls,
-            vhpiPortDecls,
-            vhpiGenericDecls,
-            vhpiConstDecls,
-            //    vhpiIndexedNames,
-            vhpiCompInstStmts,
-            vhpiBlockStmts,
-        }},
+        {vhpiRootInstK, root_options},
         {vhpiCompInstStmtK, root_options},
 
-        {vhpiGenericDeclK, sig_options = {
-            vhpiIndexedNames,
-            vhpiSelectedNames,
-        }},
+        {vhpiGenericDeclK, sig_options},
         {vhpiSigDeclK, sig_options},
         {vhpiSelectedNameK, sig_options},
         {vhpiIndexedNameK, sig_options},
         {vhpiPortDeclK, sig_options},
 
-        {vhpiCondSigAssignStmtK, simplesig_options = {
-            vhpiDecls,
-            vhpiInternalRegions,
-            vhpiSensitivitys,
-            vhpiStmts,
-        }},
+        {vhpiCondSigAssignStmtK, simplesig_options},
         {vhpiSimpleSigAssignStmtK, simplesig_options},
         {vhpiSelectSigAssignStmtK, simplesig_options},
 
-        {vhpiForGenerateK, gen_options = {
-            vhpiDecls,
-            vhpiInternalRegions,
-            vhpiSigDecls,
-            vhpiVarDecls,
-            vhpiConstDecls,
-            vhpiCompInstStmts,
-            vhpiBlockStmts,
-        }},
+        {vhpiForGenerateK, gen_options},
         {vhpiIfGenerateK, gen_options},
         {vhpiBlockStmtK, gen_options},
 

--- a/cocotb/share/lib/vpi/VpiCbHdl.cpp
+++ b/cocotb/share/lib/vpi/VpiCbHdl.cpp
@@ -561,59 +561,59 @@ VpiNextPhaseCbHdl::VpiNextPhaseCbHdl(GpiImplInterface *impl) : GpiCbHdl(impl),
 
 decltype(VpiIterator::iterate_over) VpiIterator::iterate_over = []{
     /* for reused lists */
-    std::initializer_list<int32_t> module_options;
-    std::initializer_list<int32_t> struct_options;
+    std::initializer_list<int32_t> module_options = {
+        //vpiModule,            // Aldec SEGV on mixed language
+        //vpiModuleArray,       // Aldec SEGV on mixed language
+        //vpiIODecl,            // Don't care about these
+        vpiNet,
+        vpiNetArray,
+        vpiReg,
+        vpiRegArray,
+        vpiMemory,
+        vpiIntegerVar,
+        vpiRealVar,
+        vpiRealNet,
+        vpiStructVar,
+        vpiStructNet,
+        //vpiVariables          // Aldec SEGV on plain Verilog
+        vpiNamedEvent,
+        vpiNamedEventArray,
+        vpiParameter,
+        //vpiSpecParam,         // Don't care
+        //vpiParamAssign,       // Aldec SEGV on mixed language
+        //vpiDefParam,          // Don't care
+        vpiPrimitive,
+        vpiPrimitiveArray,
+        //vpiContAssign,        // Don't care
+        vpiProcess,             // Don't care
+        vpiModPath,
+        vpiTchk,
+        vpiAttribute,
+        vpiPort,
+        vpiInternalScope,
+        //vpiInterface,         // Aldec SEGV on mixed language
+        //vpiInterfaceArray,    // Aldec SEGV on mixed language
+    };
+    std::initializer_list<int32_t> struct_options = {
+        vpiNet,
+#ifndef IUS
+        vpiNetArray,
+#endif
+        vpiReg,
+        vpiRegArray,
+        vpiMemory,
+        vpiParameter,
+        vpiPrimitive,
+        vpiPrimitiveArray,
+        vpiAttribute,
+        vpiMember,
+    };
 
     return decltype(VpiIterator::iterate_over) {
-        {vpiModule, module_options = {
-            //vpiModule,            // Aldec SEGV on mixed language
-            //vpiModuleArray,       // Aldec SEGV on mixed language
-            //vpiIODecl,            // Don't care about these
-            vpiNet,
-            vpiNetArray,
-            vpiReg,
-            vpiRegArray,
-            vpiMemory,
-            vpiIntegerVar,
-            vpiRealVar,
-            vpiRealNet,
-            vpiStructVar,
-            vpiStructNet,
-            //vpiVariables          // Aldec SEGV on plain Verilog
-            vpiNamedEvent,
-            vpiNamedEventArray,
-            vpiParameter,
-            //vpiSpecParam,         // Don't care
-            //vpiParamAssign,       // Aldec SEGV on mixed language
-            //vpiDefParam,          // Don't care
-            vpiPrimitive,
-            vpiPrimitiveArray,
-            //vpiContAssign,        // Don't care
-            vpiProcess,             // Don't care
-            vpiModPath,
-            vpiTchk,
-            vpiAttribute,
-            vpiPort,
-            vpiInternalScope,
-            //vpiInterface,         // Aldec SEGV on mixed language
-            //vpiInterfaceArray,    // Aldec SEGV on mixed language
-        }},
+        {vpiModule, module_options},
         {vpiGenScope, module_options},
 
-        {vpiStructVar, struct_options = {
-            vpiNet,
-#ifndef IUS
-            vpiNetArray,
-#endif
-            vpiReg,
-            vpiRegArray,
-            vpiMemory,
-            vpiParameter,
-            vpiPrimitive,
-            vpiPrimitiveArray,
-            vpiAttribute,
-            vpiMember,
-        }},
+        {vpiStructVar, struct_options},
         {vpiStructNet, struct_options},
 
         {vpiNet, {

